app/Http/Transformers/ActivityTransformer.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Rogue\Http\Transformers;
+
+use Rogue\Models\Event;
+use Rogue\Models\Signup;
+use League\Fractal\TransformerAbstract;
+
+class ActivityTransformer extends TransformerAbstract
+{
+    /**
+     * List of resources to automatically include
+     *
+     * @var array
+     */
+    protected $defaultIncludes = [
+        'posts', 'events',
+    ];
+
+    /**
+     * Transform resource data.
+     *
+     * @param \Rogue\Models\Signup $signup
+     * @return array
+     */
+    public function transform(Signup $signup)
+    {
+        return [
+            'signup_id' => $signup->id,
+            'signup_event_id' => $signup->event_id,
+            'submission_type' => $signup->event->submission_type,
+            'northstar_id' => $signup->northstar_id,
+            'campaign_id' => $signup->campaign_id,
+            'campaign_run_id' => $signup->campaign_run_id,
+            'quantity' => $signup->quantity,
+            'quantity_pending' => $signup->quantity_pending,
+            'why_participated' => $signup->why_participated,
+            'signup_source' => $signup->source,
+            'created_at' => $signup->created_at->toIso8601String(),
+            'updated_at' => $signup->updated_at->toIso8601String(),
+        ];
+    }
+
+    /**
+     * Include the post.
+     *
+     * @param \Rogue\Models\Signup $signup
+     * @return \League\Fractal\Resource\Collection
+     */
+    public function includePosts(Signup $signup)
+    {
+        $post = $signup->posts;
+
+        return new \League\Fractal\Resource\Collection($post, function ($post) {
```

Comment:
@chloealee as far as I know we aren't changing the responses for those (but I guess we could if we find we don't really need to return so much), but also it shouldn't matter since we definitely want to include the whole post and event in this response. So I think using the existing transformers should be fine.